jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/ideMaven/MavenRunnerUtils.kt

```diff
@@ -0,0 +1,189 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codemodernizer.ideMaven
+
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.project.Project
+import org.jetbrains.idea.maven.execution.MavenRunner
+import org.jetbrains.idea.maven.execution.MavenRunnerParameters
+import org.jetbrains.idea.maven.execution.MavenRunnerSettings
+import org.slf4j.Logger
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.services.codemodernizer.model.MavenCopyCommandsResult
+import software.aws.toolkits.jetbrains.services.codemodernizer.state.CodeTransformTelemetryState
+import software.aws.toolkits.telemetry.CodeTransformMavenBuildCommand
+import software.aws.toolkits.telemetry.CodetransformTelemetry
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+
+private fun emitMavenFailure(error: String, logger: Logger, throwable: Throwable? = null) {
+    if (throwable != null) logger.error(throwable) { error } else logger.error { error }
+    CodetransformTelemetry.mvnBuildFailed(
+        codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+        codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.IDEBundledMaven,
+        reason = error
+    )
+}
+
+fun runMavenCopyCommands(sourceFolder: File, buildlogBuilder: StringBuilder, logger: Logger, project: Project): MavenCopyCommandsResult {
+    val currentTimestamp = System.currentTimeMillis()
+    val destinationDir = Files.createTempDirectory("transformation_dependencies_temp_$currentTimestamp")
+
+    logger.info { "Executing IntelliJ bundled Maven" }
+    try {
+        // Create shared parameters
+        val transformMvnRunner = TransformMavenRunner(project)
+        val mvnSettings = MavenRunner.getInstance(project).settings.clone() // clone required to avoid editing user settings
```

Comment:
In what way could these diverge from the user settings?

Reply:
Rn they don't, when developing this I changed the values in this `mvnSettings` without cloning and unknowingly changed the actual IDE settings. This is a safeguard atm as the current code does not mutate it anymore.